lib/membrane_h264_plugin/parser/decoder_configuration_record.ex

```diff
@@ -4,26 +4,70 @@ defmodule Membrane.H264.Parser.DecoderConfigurationRecord do
 
   The structure of the record is described in section 5.2.4.1.1 of MPEG-4 part 15 (ISO/IEC 14496-15).
   """
+
+  alias Membrane.H264.Parser
+
   @enforce_keys [
-    :sps,
-    :pps,
+    :spss,
+    :ppss,
     :avc_profile_indication,
     :avc_level,
     :profile_compatibility,
-    :length_size_minus_one
+    :nalu_length_size
   ]
   defstruct @enforce_keys
 
   @typedoc "Structure representing the Decoder Configuartion Record"
   @type t() :: %__MODULE__{
-          sps: [binary()],
-          pps: [binary()],
+          spss: [binary()],
+          ppss: [binary()],
           avc_profile_indication: non_neg_integer(),
           profile_compatibility: non_neg_integer(),
           avc_level: non_neg_integer(),
-          length_size_minus_one: non_neg_integer()
+          nalu_length_size: non_neg_integer()
         }
 
+  @doc """
+  Generates a DCR based on given PPSs and SPSs.
+  """
+  @spec generate([binary()], [binary()], Parser.stream_structure()) ::
+          binary() | nil
+  def generate(spss, ppss, {avc, nalu_length_size}) do
+    sps_common_parameters =
+      Enum.map(spss, fn <<_idc_and_type, profile, compatibility, level, _rest::binary>> ->
+        <<profile, compatibility, level>>
+      end)
+      |> Enum.uniq()
+
+    if length(sps_common_parameters) > 1 do
+      raise("SPS parameters should be the same for all sets but are different")
```

Comment:
I don't think we should raise here - AFAIK there might be SPS'es with different profiles in the same DCR

Reply:
true, we should just take the profiles and stuff from the newest one